util-cgroups/tests/integration.rs

```diff
@@ -0,0 +1,892 @@
+    let file_path_cpu = dir_path.join("cpu.stat");
+    let mut file_cpu = File::create(file_path_cpu.clone()).unwrap();
+    cpu_stat_mock_file.write_to_file(&mut file_cpu).unwrap();
+    assert!(file_path_cpu.exists());
+
+    let cpt: std::sync::Arc<AtomicU8> = std::sync::Arc::new(AtomicU8::new(0));
+    let ac_out = std::sync::Arc::clone(&cpt);
+    let root_clo = std::sync::Arc::new(root.clone());
+    let shared_root = std::sync::Arc::clone(&root_clo);
+
+    let f1 = callback(move |cgroups| {
+        for cgroup in cgroups {
+            let root_str = shared_root.file_name().unwrap().to_str().unwrap();
+            let vec_child = vec![root_str, "William_Parry"];
+            assert!(vec_child.iter().any(|elm| { cgroup.fs_path().ends_with(elm) }));
+            assert!(cgroup.fs_path().starts_with(shared_root.as_path()));
+            ac_out.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+            assert!(cgroup.fs_path().join("cpu.stat").exists());
+            assert!(!cgroup.fs_path().join("memory.stat").exists());
+        }
+        Ok(())
+    });
+    let f2 = callback(|_a| Ok(()));
+    let handler = ClosureCallbacks {
+        on_cgroups_created: f1,
+        on_cgroups_removed: f2,
+    };
+    let detector = CgroupDetector::new(hierarchy_cpu, Config::default(), handler);
+    assert!(detector.is_ok());
+
+    thread::sleep(Duration::from_secs(5));
+    assert_eq!(2, cpt.load(std::sync::atomic::Ordering::SeqCst));
+}
+
+#[test]
+fn remove_cgroup_on_exec_v2() {
+    let root_unwraped = tempdir().unwrap();
+    let root = root_unwraped.path().to_path_buf();
+    create_files_cgroupv2(&root).unwrap();
+    let version = CgroupVersion::V2;
+
+    let hierarchy_cpu = CgroupHierarchy::manually_unchecked(root.clone(), version, vec!["cpuacct"]);
+    let sub_root_p = create_folder(&root, "Pantalaimon", hierarchy_cpu.clone()).unwrap();
+    let cpt: std::sync::Arc<AtomicU8> = std::sync::Arc::new(AtomicU8::new(0));
+    let ac_out = std::sync::Arc::clone(&cpt);
+    let root_clo = std::sync::Arc::new(root.clone());
+    let shared_root = std::sync::Arc::clone(&root_clo);
+
+    let f1 = callback(move |cgroups| {
+        for cgroup in cgroups {
+            let root_str = shared_root.file_name().unwrap().to_str().unwrap();
+            let vec_child = vec![root_str, "Pantalaimon"];
+            assert!(vec_child.iter().any(|elm| { cgroup.fs_path().ends_with(elm) }));
+            assert!(cgroup.fs_path().starts_with(shared_root.to_path_buf()));
+            ac_out.fetch_add(1, std::sync::atomic::Ordering::SeqCst);
+        }
+        Ok(())
+    });
+    let f2 = callback(|_a| Ok(()));
+    let handler = ClosureCallbacks {
+        on_cgroups_created: f1,
+        on_cgroups_removed: f2,
+    };
+    let detector = CgroupDetector::new(hierarchy_cpu.clone(), Config::default(), handler);
+    assert!(detector.is_ok());
+    assert!(fs::remove_dir_all(sub_root_p).is_ok());
+
+    thread::sleep(Duration::from_secs(5));
```

Comment:
It would be great if this duration was a constant, so that we can change it everywhere at once if we want to.